web-crypto/sign-verify/ed25519.js

```diff
@@ -0,0 +1,88 @@
+(() => {
+
+  /*
+  Store the calculated signature here, so we can verify it later.
+  */
+  let signature;
+
+  /*
+  Fetch the contents of the "message" textbox, and encode it
+  in a form we can use for sign operation.
+  */
+  function getMessageEncoding() {
+    const messageBox = document.querySelector("#ed25519-message");
+    let message = messageBox.value;
+    let enc = new TextEncoder();
+    return enc.encode(message);
+  }
+
+  /*
+  Get the encoded message-to-sign, sign it and display a representation
+  of the first part of it in the "signature" element.
+  */
+  async function signMessage(privateKey) {
+    const signatureValue = document.querySelector(".ed25519 .signature-value");
+    signatureValue.classList.remove("valid", "invalid");
+
+    let encoded = getMessageEncoding();
+    signature = await window.crypto.subtle.sign(
+      {
+        name: "Ed25519"
+      },
+      privateKey,
+      encoded
+    );
```

Comment:
It's not strictly necessary to pass an object as the algorithm ID in this case, since there's only the name of the algorithm to be passed:

```suggestion
    signature = await window.crypto.subtle.sign("Ed25519", privateKey, encoded);
```

(also, the `window.` isn't strictly necessary, but I see all the other examples have that as well)